### PR TITLE
1095 add https support

### PR DESCRIPTION
**Pull request by grzanka**
This is needed to get working key cloak integration, see https://github.com/yaptide/ui/pull/1086

### CODE REVIEW

**Comment by grzanka** [comment]
image

backend needs to be served over https as well

**Comment by grzanka** [comment]
Here is a prototype deploy script
https://github.com/yaptide/deploy/pull/33/files

File: Dockerfile [hunk 1]
```diff
@@ -17,6 +29,10 @@ RUN apt-get update && apt-get install -y --no-install-recommends python3-pip pyt
 # Directory where the app is installed and run.
 WORKDIR /usr/src/app
 
+# Copy SSL certificate from the cert-gen stage
+COPY --from=cert-gen /certs/server.key /usr/src/app/server.key
+COPY --from=cert-gen /certs/server.crt /usr/src/app/server.crt
```
Comment (hendzeld) [inline Dockerfile:34]: Do we need to copy these files, are they required to build the APP?
Maybe they should be copied only in the last state.

File: Dockerfile [hunk 2]
```diff
@@ -54,7 +70,22 @@ RUN npm run fix-web-dev
 
 # Stage 2: Serve the app using Nginx.
 FROM nginx:alpine
+
+# Remove the default Nginx configuration
+RUN rm /etc/nginx/conf.d/default.conf
+
+# Copy SSL certificate from the build stage
+COPY --from=build /usr/src/app/server.key /etc/nginx/conf.d/server.key
+COPY --from=build /usr/src/app/server.crt /etc/nginx/conf.d/server.crt
```
Comment (hendzeld) [inline Dockerfile:79]: Related to previous commend

```suggestion
COPY --from=cert-gen /certs /etc/nginx/conf.d
```

File: Dockerfile [hunk 3]
```diff
@@ -54,7 +70,22 @@ RUN npm run fix-web-dev
 
 # Stage 2: Serve the app using Nginx.
```
Comment (hendzeld) [inline Dockerfile:71]:
```suggestion
# Stage 3: Serve the app using Nginx.
```

**Review by hendzeld — COMMENTED** [review]
Small suggestions.

Also, do we want to update README?

There is a section about building and running with docker.
https://github.com/yaptide/ui#building-the-app-using-the-dockerfile

File: Dockerfile (hunk 1 above)
Comment (grzanka) [reply Dockerfile:34]: Good point, if docker allows we may as well copy them from stage 1 to stage 3 directly.

Comment (hendzeld) [reply Dockerfile:34]: I checked it locally and it works, added a suggestion in another comment.

So we can remove these lines if they are not required
```suggestion
```

**Review by hendzeld — APPROVED** [review]
LGTM

**Pull request merged**